### ISSUE TITLE
flutter upgrade crash

### ISSUE THREAD

**Issue by dcov** [flutter/flutter#30203]
`flutter upgrade` crashes while building the flutter tool.
`flutter upgrade -v` output:
```
[dcov@dev ~]$ flutter upgrade -v
[   +4 ms] executing: [/home/dcov/flutter/] git log -n 1 --pretty=format:%H
[  +10 ms] Exit code 0 from: git log -n 1 --pretty=format:%H
[        ] ffbb335e35ae7416be98b72779baae6c9bd25b3b
[        ] executing: [/home/dcov/flutter/] git describe --match v*.*.* --first-parent --long --tags
[  +13 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.4.6-7-gffbb335e3
[   +3 ms] executing: [/home/dcov/flutter/] git rev-parse --abbrev-ref HEAD
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[   +2 ms] executing: [/home/dcov/flutter/] git describe --match v*.*.* --first-parent --long --tags
[   +9 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.4.6-7-gffbb335e3
[   +2 ms] executing: [/home/dcov/flutter/] git rev-parse @{u}
[  +16 ms] ffbb335e35ae7416be98b72779baae6c9bd25b3b
[        ] executing: [/home/dcov/flutter/] git reset --hard v1.4.6
[  +26 ms] HEAD is now at ec93c87c8 Move spinner `_defaultSlowWarning` message to a new line (#30071)
[   +1 ms] Upgrading Flutter from /home/dcov/flutter...
[   +2 ms] executing: [/home/dcov/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/home/dcov/flutter/] git ls-remote --get-url origin
[   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter
[   +3 ms] executing: [/home/dcov/flutter/] git pull --ff
[ +446 ms] Updating ec93c87c8..ffbb335e3
[   +4 ms]  .../bin/src/main/res/mipmap-hdpi/ic_launcher.png   | Bin 0 -> 544 bytes
[        ]  .../bin/src/main/res/mipmap-mdpi/ic_launcher.png   | Bin 0 -> 442 bytes
[        ]  .../bin/src/main/res/mipmap-xhdpi/ic_launcher.png  | Bin 0 -> 721 bytes
[        ]  .../bin/src/main/res/mipmap-xxhdpi/ic_launcher.png | Bin 0 -> 1031 bytes
[        ]  .../src/main/res/mipmap-xxxhdpi/ic_launcher.png    | Bin 0 -> 1443 bytes
[        ]  .../app/src/main/res/mipmap-hdpi/ic_launcher.png   | Bin 0 -> 544 bytes
[        ]  .../app/src/main/res/mipmap-mdpi/ic_launcher.png   | Bin 0 -> 442 bytes
[        ]  .../app/src/main/res/mipmap-xhdpi/ic_launcher.png  | Bin 0 -> 721 bytes
[        ]  .../app/src/main/res/mipmap-xxhdpi/ic_launcher.png | Bin 0 -> 1031 bytes
[        ]  .../src/main/res/mipmap-xxxhdpi/ic_launcher.png    | Bin 0 -> 1443 bytes
[   +1 ms]  68 files changed, 906 insertions(+), 277 deletions(-)
[   +4 ms] Upgrading engine...
[        ] executing: [/home/dcov/flutter/] bin/flutter --no-color --no-version-check precache
[  +33 ms] Building flutter tool...

===== CRASH =====
si_signo=Segmentation fault(11), si_code=128, si_addr=(nil)
version=2.2.1-dev.2.0.flutter-None (Thu Mar 28 00:48:30 2019 +0000) on "linux_x64"
thread=3781, isolate=flutter_tools.snapshot:main()(0x2920e00)
Stack dump aborted because InitialRegisterCheck failed.
/home/dcov/flutter/bin/flutter: line 177:  3645 Aborted                 (core dumped) "$DART" $FLUTTER_TOOL_ARGS "$SNAPSHOT_PATH" "$@"
```
`flutter doctor -v` output:
```
[dcov@dev ~]$ flutter doctor -v
[✓] Flutter (Channel master, v1.4.7-pre.7, on Linux, locale en_US.UTF-8)
    • Flutter version 1.4.7-pre.7 at /home/dcov/flutter
    • Framework revision ffbb335e35 (69 minutes ago), 2019-03-29 10:49:07 -0700
    • Engine revision c4d14a0071
    • Dart version 2.2.1 (build 2.2.1-dev.2.0 None)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/dcov/android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /home/dcov/android
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] Android Studio (version 3.2)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 29.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] Connected device (1 available)
    • SM G930T • 5c42024b • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
```

**Comment by jonahwilliams** [comment]
@dcov what happens if you run `flutter` again? Does it continue to crash, or does it build correctly the second time

**Comment by dcov** [comment]
I'm assuming you meant if I run `flutter upgrade` again? Because it still crashes when building the flutter tool.

**Comment by jonahwilliams** [comment]
No, just `flutter` by itself. This should force a the tool to be built.

**Comment by dcov** [comment]
Ah ok my mistake, if I run `flutter` by itself it prints out the list of available commands.

**Comment by jonahwilliams** [comment]
I'm disabling the snapshotting logic in https://github.com/flutter/flutter/pull/30204 to fix this. Did you end up on the correct version, or did the upgrade fail before it completed?

**Comment by dcov** [comment]
I'm not sure, it downloads and updates all of the packages fine, and only fails when building the flutter tool. I can still use the tool but I'm assuming it's the previous version?
This is what `flutter --version` shows:
```
[dcov@dev ~]$ flutter --version
Flutter 1.4.7-pre.7 • channel master • https://github.com/flutter/flutter
Framework • revision ffbb335e35 (2 hours ago) • 2019-03-29 10:49:07 -0700
Engine • revision c4d14a0071
Tools • Dart 2.2.1 (build 2.2.1-dev.2.0 None)
```

**Comment by jonahwilliams** [comment]
Fixed via #30204

**Comment by mustufaalighiya** [comment]
Launching lib\main.dart on Windows in debug mode...
Building Windows application...

===== CRASH =====
ExceptionCode=-1073741819, ExceptionFlags=0, ExceptionAddress=00007FF748943B22
version=2.13.0-77.0.dev (dev) (Thu Feb 25 00:41:11 2021 -0800) on "windows_x64"
pid=10948, thread=9760, isolate_group=main(0000014E5795B530), isolate=main(0000014E5796F3C0)
isolate_instructions=14e5abe0000, vm_instructions=7ff74886fa30
Stack dump aborted because InitialRegisterCheck failed.
